modules/custom/az_migration/src/Plugin/migrate/process/DefaultLangcode.php

```diff
@@ -0,0 +1,81 @@
+<?php
+
+namespace Drupal\az_migration\Plugin\migrate\process;
+
+use Drupal\Core\Language\LanguageInterface;
+use Drupal\Core\Language\LanguageManagerInterface;
+use Drupal\migrate\MigrateExecutableInterface;
+use Drupal\migrate\ProcessPluginBase;
+use Drupal\migrate\Row;
+use Symfony\Component\DependencyInjection\ContainerInterface;
+
+/**
+ * Provides a 'az_default_langcode' migrate process plugin.
+ *
+ * @MigrateProcessPlugin(
+ *  id = "az_default_langcode"
+ * )
+ *
+ * This plugin can be used in a migration to set the language code of the
+ * destination entity. If the incoming language value is 'und', the plugin
+ * will use the destination site's default langcode. Otherwise, it will use
+ * the incoming language value.
+ *
+ * Example usage:
+ *
+ * @code
+ * process:
+ *   langcode:
+ *     plugin: az_default_langcode
+ *     source: source_langcode
+ * @endcode
+ */
+class DefaultLangcode extends ProcessPluginBase {
+
+  /**
+   * The language manager service.
+   *
+   * @var \Drupal\Core\Language\LanguageManagerInterface
+   */
+  protected $languageManager;
+
+  /**
+   * Constructs a DefaultLangcode object.
+   *
+   * @param array $configuration
+   *   A configuration array containing information about the plugin instance.
+   * @param string $plugin_id
+   *   The plugin_id for the plugin instance.
+   * @param mixed $plugin_definition
+   *   The plugin implementation definition.
+   * @param \Drupal\Core\Language\LanguageManagerInterface $language_manager
+   *   The language manager service.
+   */
+  public function __construct(array $configuration, $plugin_id, $plugin_definition, LanguageManagerInterface $language_manager) {
+    parent::__construct($configuration, $plugin_id, $plugin_definition);
+    $this->languageManager = $language_manager;
+  }
+
+  /**
+   * {@inheritdoc}
+   */
+  public static function create(ContainerInterface $container, array $configuration, $plugin_id, $plugin_definition) {
```

Comment:
Currently, the PR is producing an error in our migration tests:

> TypeError: Drupal\az_migration\Plugin\migrate\process\DefaultLangcode::__construct(): Argument #4 ($language_manager) must be of type Drupal\Core\Language\LanguageManagerInterface, Drupal\migrate\Plugin\Migration given, called in /src/az-quickstart-scaffolding/web/core/modules/migrate/src/Plugin/MigratePluginManager.php on line 60 in Drupal\az_migration\Plugin\migrate\process\DefaultLangcode->__construct() (line 54 of /src/az-quickstart-scaffolding/web/profiles/custom/az_quickstart/modules/custom/az_migration/src/Plugin/migrate/process/DefaultLangcode.php).
> [error]  TypeError: Drupal\az_migration\Plugin\migrate\process\DefaultLangcode::__construct(): Argument #4 ($language_manager) must be of type Drupal\Core\Language\LanguageManagerInterface, Drupal\migrate\Plugin\Migration given, called in /src/az-quickstart-scaffolding/web/core/modules/migrate/src/Plugin/MigratePluginManager.php on line 60 in Drupal\az_migration\Plugin\migrate\process\DefaultLangcode->__construct() (line 54 of /src/az-quickstart-scaffolding/web/profiles/custom/az_quickstart/modules/custom/az_migration/src/Plugin/migrate/process/DefaultLangcode.php) #0 /src/az-quickstart-scaffolding/web/core/modules/migrate/src/Plugin/MigratePluginManager.php(60): Drupal\az_migration\Plugin\migrate\process\DefaultLangcode->__construct()

Perhaps this change is needed?

```suggestion
  public static function create(ContainerInterface $container, array $configuration, $plugin_id, $plugin_definition, MigrationInterface $migration = NULL) {
```

Reply:
Thanks @bberndt-uaz